src/vsr/journal.zig

```diff
@@ -515,27 +515,19 @@ pub fn JournalType(comptime Replica: type, comptime Storage: type) type {
             return op;
         }
 
-        pub fn has(journal: *const Journal, header: *const Header.Prepare) bool {
+        pub fn has_header(journal: *const Journal, header: *const Header.Prepare) bool {
             assert(journal.status == .recovered);
             assert(header.command == .prepare);
             assert(header.operation != .reserved);
 
-            const slot = journal.slot_for_op(header.op);
-            const existing = &journal.headers[slot.index];
-            if (existing.operation == .reserved) {
-                return false;
+            if (journal.header_with_op_and_checksum(header.op, header.checksum)) |_| {
+                return true;
```

Comment:
Took the opportunity to also refactor `has_header`, I observed that its logic (barring the `assert(existing.checksum_body == header.checksum_body);`) was _exactly_ the same as `header_with_op_and_checksum`!